addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1492,6 +1493,183 @@ options.registry.gallery = options.Class.extend({
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _setActive: function () {
+        this._super(...arguments);
+        this.$el.find('[data-display]').removeClass('active')
+            .filter('[data-display=' + this.$target[0].dataset.display + ']').addClass('active');
+        this.$el.find('[data-end-action]').removeClass('active')
+            .filter('[data-end-action=' + this.$target[0].dataset.endAction + ']').addClass('active');
+        this.$el.find('[data-layout]').removeClass('active')
+            .filter('[data-layout=' + this.$target[0].dataset.layout + ']').addClass('active');
+
+        this.$el.find('[data-end-time] > input').datetimepicker('date', moment(this.endTimeValue * 1000));
+        this.$el.find('[data-redirect-url] > input').val(this.redirectUrlValue);
+    },
+    /**
+     * @private
+     */
+    _updateUI: function () {
+        this._super(...arguments);
+        var dataset = this.$target[0].dataset;
+
+        // End Action UI
+        this.$el.find('[data-redirect-url]')
+            .toggleClass('d-none', dataset.endAction !== 'redirect');
+        this.$el.find('.toggle-edit-message')
+            .toggleClass('d-none', dataset.endAction !== 'message');
+        this.$el.find('[data-toggle-class="hide-countdown"]')
+            .toggleClass('d-none', dataset.endAction === 'nothing');
+
+        // End Message UI
+        this._updateUIEndMessage();
+
+        // Advanced UI
+        var showAdvanced = dataset.layout === 'advanced';
+        this.$el.find('[data-attribute-name="shape"]')
+            .toggleClass('d-none', !showAdvanced);
+        this.$el.find('[data-attribute-name="shapeBackground"], [data-attribute-name="progressBarStyle"]')
+            .toggleClass('d-none', !showAdvanced || dataset.shape === 'none');
+        this.$el.find('[data-attribute-name="progressBarWeight"]')
+            .toggleClass('d-none', !showAdvanced || dataset.progressBarStyle === 'none');
+    },
+    /**
+     * Shows or hides the countdown and its message depending of the chosen
+     * snippet options.
+     *
+     * @private
+     */
+    _updateUIEndMessage: function () {
+        var t = this.$target;
+        t.find('.s_countdown_canvas_wrapper')
+            .toggleClass("d-none", t.is(".show-end-message.hide-countdown"));
+        t.find('.s_countdown_end_message').toggleClass("d-none", !t.hasClass("show-end-message"));
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onToggleEndMessageClick: function () {
+        var a = this.$el.find(".toggle-edit-message"),
+            i = a.find("i"),
+            t = this.$target;
```

Comment:
Did al code this ? :p jQuery variables should at least start with `$`. Normally we also don't use multiple variable assignement, it's less clear for no reason to me. Most of the time you should now use `const` instead of `var`, sometimes `let`, practically never `var`.

Reply:
No but you are right that it wasn't me.
Since it is an (very) unusual feature, I copied it from [where it was created](https://github.com/odoo/design-themes/blame/ef626bcd5cc0cf9f141348965243ff61cfef9723/theme_common/static/src/js/s_animated_boxes_editor.js#L47).

Changed, kept the shortcut variable name but added `$` to it.